hs/src/Reach/Eval/Core.hs

```diff
@@ -4351,7 +4369,7 @@ doApiCall lhs (ApiCallRec{..}) = do
   let returnVal = jidg "rng"
   let returnLVal = jidg "rngl"
   let interactOut = jsCall a (mkIdDot a ["interact", "out"]) [dom, returnLVal]
-  let doLog = jsCall a (jid ".emitLog") [ jidg "rng" ]
+  let doLog = jsCall a (jid ".emitLog") [ jidg "rng", who_str ]
```

Comment:
Why do logs need to know the name of the participant?

Reply:
(i.e. you could just allocate a new variable and associate it with the participant)

Reply:
makes sense, Ill add this

Reply:
I'm actually not sure I understand because if you have a map of `int - slval` you'd still have to expand it in `doEmitLog` and expect a participant. Unless you mean, handle the expansion in solidity generation, so the map would have to be passed through the `program` field?

Reply:
I don't know if this is a good idea.

What I mean is that `EmitLog` already has a `DLVar`, which is the one that it is outputting. For example, it says `DLE_EmitLog _ _ v22` and that `v22` is tied to this log entry. We use it for the event value in Solidity and in the ALGO log header. What you could do is make it so that `doEmitLog` (in `Core`) associates `v22` with `Flower_girl` (i.e. each API has an attribute of "What is the variable that my output comes from?")

Then, every logged value gets put inside the memory table for potential returning.

Why could this be better? It is a more "general" feature rather than tying things specifically to API names.

Reply:
I think the issue with this approach is that every `emitLog` used by APIs is generated via JavaScript and we cannot know the DLVar associated with the log entry when generating the JavaScript. So, any association made between `v22` and `Flower_girl` would have to be as a result of passing arguments to the `emitLog` function. If I'm still not understanding this, maybe you can provide an example of something that would utilize the generalness of what you're describing so I can better reason about the compilation path..

Reply:
I don't know what you mean by generated via JavaScript... these emit logs are generated inside of Core and they get compiled to Solidity & JS & TEAL

Reply:
Sorry,  I meant we generate Javascript, `.emitLog(..)`, in Core that is later evaluated